mu/modes/esp.py

```diff
@@ -38,15 +38,25 @@ class ESPMode(MicroPythonMode):
     icon = "esp"
     fs = None
 
-    # There are many boards which use ESP microcontrollers but they often use
-    # the same USB / serial chips (which actually define the Vendor ID and
-    # Product ID for the connected devices.
+    # The below list defines the supported devices, however, many
+    # devices are using the exact same FTDI USB-interface, with vendor
+    # ID 0x403 without reporting their own VID/PID
+
+    # In some instances we can recognize the device not on VID/PID,
+    # but on manufacturer ID, that's what the third column is for.
+    # These more specific device specifications, should be listed
+    # before the generic FTDI VID/PID's
     valid_boards = [
         # VID  , PID,    Manufacturer string, Device name
         (0x1A86, 0x7523, None, "HL-340"),
         (0x10C4, 0xEA60, None, "CP210x"),
-        (0x0403, 0x6015, None, "Sparkfun ESP32 Thing"),
         (0x0403, 0x6001, "M5STACK Inc.", "M5Stack ESP32 device"),
+        (0x0403, 0x6001, None, None),  # FT232/FT245 (XinaBox CW01, CW02)
```

Comment:
@dybber is this going to be recognised as a M5Stack?

Reply:
Yes, the idea is that if a device matches multiple lines, the name reported will be the first in the list. This means that a device with VID/PID 0x0403 0x6001, and "M5Stack Inc." as manufacturer will be shown as "M5Stack ESP32 device", whereas anyother device with the same VID/PID will just be shown as "ESP MicroPython"

Here is a screenshot where I had a M5StickC device and a Sparkfun ESP32 Thing connected simultaneously (VID/PID 0x403 0x6015):

![Screenshot 2020-09-05 at 18 59 55](https://user-images.githubusercontent.com/55204/92310008-666b2c80-efaa-11ea-8ac7-c36465e37641.png)

Reply:
I guess I was surprised the manufacturer string was configured in FTDI chips.
So then in an M5Stack ESP32 the [QSerialPortInfo.manufacturer()](https://doc.qt.io/qt-5/qserialportinfo.html#manufacturer) returns "M5STACK Inc."?

Reply:
Yes, that's a bit confusing, but I guess it's just because they don't want to pay the $5000 to get their own Vendor ID.

Reply:
So the FTDI chips in the M5 Stack have the USB manufacturer string set to "M5STACK Inc."?
I don't have one to check, so just wanted to make sure that is the case.

Reply:
Exactly, and otherwise (if they change that in the future), it will just show up as ESP MicroPython.

Reply:
Great, thanks Martin!